projects/igniteui-angular/src/lib/grids/common/pipes.ts

```diff
@@ -229,7 +229,7 @@ export class IgxGridRowPinningPipe implements PipeTransform {
 
         if (grid.hasPinnedRecords && isPinned) {
             const result = collection.filter(rec => grid.isRecordPinned(rec));
-            result.sort((rec1, rec2) => grid.pinRecordIndex(rec1) - grid.pinRecordIndex(rec2));
+            result.sort((rec1, rec2) => grid.getPinedRecordIndex(rec1) - grid.getPinedRecordIndex(rec2));
```

Comment:
getPinedRecordIndex and setPinnedRecordGhostIndex will probably have to be overwritten for the treegrid as its data records are of different type - ITreeGridRecord. Hence these methods will not work as expected for them.